### ISSUE TITLE
An indicator of whether the frame is 'broadcast compatible'

### ISSUE THREAD

**Issue by rawles** [rawles/teletext-editor#62]
As frames find their way to places like Teefax, we would like to encourage designers to keep rows 0 and 24 free for broadcast systems to replace the header and insert Fastext links. At the moment the editor does nothing to do this besides the grid (Esc-X).

@peterkvt80 suggested a flag warning when a frame is not 100% broadcast compliant. Let's put this in the status bar. Initially I think we could make this three colours, like a traffic light.

Red: You are using row 0, which will be overwritten by an inserter.
Yellow: You are using row 24, which will be overwritten by Fastext links. We could also use this to mean 'you are using foreground black', but that's totally normal and usual on the teletext services of other countries.
Green: Rows 0 and 24 are free (and maybe, foreground back hasn't been used), everything is groovy.

Does this seem sensible to everyone?

[Paging @ZXGuesser and @pjfdirect who knows about these things too]

**Comment by pjfdirect** [comment]
Sounds good to me.

I have a 'teleletext design' post in the works sharing my design goodies with Dan and the general public, plus links to editors and such - I'm happy to echo those guidelines.

Some who read it might be Teefax bound but others might just be footloose pixel artists or doing 'not necessarily teletext' things like Alexi Sayle's thing below so it may fast track them into the distinctions.

![ceefax by sayle](https://cloud.githubusercontent.com/assets/23056210/20526922/20b5ef2e-b0be-11e6-8d0b-bf6bab1f1f71.png)

**Comment by rawles** [comment]
Great, we can add that to the list of resources for people starting out, alongside the how-to guide and stuff like that. 

What are your opinions on whether foreground black should be considered a 'yellow light'?

**Comment by ZXGuesser** [comment]
Sounds sensible. I don't have much to add besides that when it comes to leaving Row 24 free, that row is not used in systems that use Table Of Pages navigation instead of FLOF (fasttext).
Where fasttext is in use it's more a case of the fast text links being an decision made by the person running the service rather than the author of a particular page - or they may be the same person.

Reminding people that row 24 should only be used for the prompts for fasttext links is a good idea though. While you can ignore the FLOF code of practice and put anything there the spec does warn that it's not advisable to allow enhancement objects to spread into row 24 because some decoders display row 24 below row 11 when in "page expansion" mode (i.e. zoomed). Though it doesn't mention this in regard to using row 24 for things other than in conformity with the FLOF code of practice I would think the same argument applies against putting general content on that row too.

**Comment by ZXGuesser** [comment]
Another thought is that the distinctions of row 0 and row 24 being "special" in teletext is irrelevant entirely if you are using edit.tf to design Mode 7 artwork for a BBC microcomputer.

**Comment by rawles** [comment]
Thanks for all the very detailed information! Definitely a good point about the BBC micro stuff. By the way, the editor doesn't implement BBC Micro display properly, e.g. how the top and bottom of a double height line can be, for example, different colours. Anyway, this is why I like Peter's suggestion - a little flag or traffic light is subtle, and can be ignored if the designer isn't making a teletext frame. Likewise a subtle box that can be turned off.

**Comment by peterkvt80** [comment]
I don't think that foreground black is that bad a thing as everything should support it these days. What would be interesting would be a "Level" indicator which is set to the level of the highest feature used. So for the purists, a page within the original IBA spec. would be 1.0. Black foreground would be 2.5 etc.

**Comment by rawles** [comment]
I've just added this and merged it in - there is a little traffic light on the bottom right of the screen. By default, there are subtle grid lines showing the teletext area. Have a look - do you think it's ok but a bit much?

**Comment by rawles** [comment]
The grid also disappears temporarily when the status bar is hidden (bug #61)

**Comment by rawles** [comment]
A case which this doesn't cover, but which would be easy to detect: if row 23 has double height characters on it, then teletext receivers will render those characters in row 24. Does that turn our light yellow? I'd say probably yes.

**Comment by rawles** [comment]
The light also goes *bright* red if any of the first eight characters are non-blank! :)

**Comment by ZXGuesser** [comment]
I would say it should go red. As per the notes on double height and size in Table 26 of The Spec my editor forbids setting double size in rows 23 or 24 (and strips them on import). The same for row zero but that is stripped entirely anyway in my case.

**Comment by peterkvt80** [comment]
Funny you should mention double height in row zero. wxTED and my TV supports that.

**Comment by ZXGuesser** [comment]
Well the specification doesn't seem to say anywhere that decoders shouldn't accept it, only that broadcasters shouldn't transmit it 😄 

"If a decoder receives an illegal combination, it may chose, for example, to display a partial
character, to reduce the size or to ignore the character completely."

**Comment by irrelevantdotcom** [comment]
Here's another frame size limitation -
.
Prestel imposed a limit of 920 characters per frame, including at least 44 for line 1. Control codes took up 2 each. Lines were terminated by CRLF (unless they were already 39/40 characters wide!) so could be less than 40 characters.  This did mean that if you do complicated full-screen graphics you could easilly run out of characters to do them in!  It did encourage some extra creativity... (judicious use of double-height when you might otherwise not have thought to use it could save an entire line of data!)

Probably not an issue for any applications these days - most file formats I have come across store a straight 22 or 23x40 image array, but it's something that sprung to mind!
